tools/build/Library/InstallUnpacker/classes/ConfigurationValidator.php

```diff
@@ -0,0 +1,155 @@
+     * by performing multiple file operations
+     *
+     * @return string[]
+     */
+    public function testFilesystemCapabilities()
+    {
+        // choose a random available directory name
+        $dirPath = $this->getRandomDirectoryPath();
+
+        $randomNameIsActuallyAlreadyUsed = (is_dir($dirPath) || file_exists($dirPath));
+        if ($randomNameIsActuallyAlreadyUsed) {
+            // retry another random directory name
+            $dirPath = $this->getRandomDirectoryPath();
+
+            $randomNameIsActuallyAlreadyUsed = (is_dir($dirPath) || file_exists($dirPath));
+            if ($randomNameIsActuallyAlreadyUsed) {
+                throw new \RuntimeException('Failed to find available directory name');
+            }
+        }
+
+        // create directory
+        $createDirectoryResult = @mkdir($dirPath);
+        if (false === $createDirectoryResult) {
+            return ['Cannot create directories'];
+        }
+
+        // create file in it
+        $fileCreationTestPath = $dirPath . DIRECTORY_SEPARATOR . 'test-file.php';
+        $createFileResult = @file_put_contents($fileCreationTestPath, "<?php echo 'Hello world !';");
+        if (false === $createFileResult) {
+            return ['Cannot write files'];
+        }
+
+        // download a file from github in directory
+        $downloadTestPath = $dirPath . DIRECTORY_SEPARATOR . 'test-download.txt';
+        // @todo: use another file from the network ?
+        $target = 'https://raw.githubusercontent.com/PrestaShop/PrestaShop/develop/robots.txt';
+        $downloadFileResult = @file_put_contents($downloadTestPath, Download::fileGetContents($target));
+        if (false === $downloadFileResult) {
+            return ['Cannot download files from network'];
+        }
+
+        // move a file from test directory into root directory
+        $fileMoveTestPath = __DIR__ . DIRECTORY_SEPARATOR . 'test-move.php';
+        $moveResult = @rename($fileCreationTestPath, $fileMoveTestPath);
+        if (false === $moveResult) {
+            return ['Cannot move files into prestashop root directory'];
+        }
+
+        // delete test file in root directory
+        $deleteFileResult = unlink($fileMoveTestPath);
+        if (false === $deleteFileResult) {
+            return ['Cannot delete files in prestashop root directory'];
+        }
+
+        // delete test directory
+        $deleteDirectoryContentResult = array_map('unlink', glob($dirPath . DIRECTORY_SEPARATOR . "*.*"));
+        $deleteDirectoryResult = @rmdir($dirPath);
+        if ((false === $deleteDirectoryContentResult) || (false === $deleteDirectoryResult)) {
+            return ['Cannot delete directories in prestashop root directory'];
+        }
+    }
+
+    /**
+     * @return string
+     */
+    private function getRandomDirectoryPath()
+    {
+        $randomDirectoryName = 'test-' . rand(0, 999);
+        $dirPath = __DIR__ . DIRECTORY_SEPARATOR . $randomDirectoryName;
```

Comment:
You expect to create new sub directories in `tools/build/Library/InstallUnpacker/classes/`?
I'm not sure this is a good idea, you could use the cache dir instead.

Reply:
This code is not run. In a "build" step, all the php classes in `classes` will be inlined into a single `index.php` file that is then zipped. This is what merchands and developers when they download "PrestaShop Download".

This `index.php` file is then run in the main/root directory of the eshop by either the merchant or the developer in order to perform the install. So at runtime, there is only 1 folder with 3 files in it (the 3 files in the zip): `prestashop.zip`, `Install_PrestaShop.html` and `index.php`.